File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        i = 0
        j = 1
        profit = 0
        maxi = 0

        while j<len(prices):
            if prices[i]>prices[j]:
                i = j
            else:
                profit = prices[j]-prices[i]
                maxi = max(maxi, profit)
            j+=1
        return maxi

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {

        int i = 0;
        int j = 1;
        int max = 0;
        int diff;
        while(j<prices.length){
            if(prices[i]<prices[j]){
                diff = prices[j]-prices[i];
                max = Math.max(diff, max);
                j++;
            }
            else{
                i = j;
                j++;
            }
            


        }

        return max;
    }
}